### ISSUE TITLE
Support portable pdb files

### ISSUE THREAD

**Issue by jp7677** [OpenCover/opencover#610]
Please provide the following information when submitting an issue, where appropriate replace the `[ ]` with a `[X]`
### My Framework
- [ ] .NET 2
- [ ] .NET 3.5
- [ ] .NET 4
- [ ] .NET 4.5
- [ ] .NET 4.6
- [ ] .NET 4.6.1
- [ ] .NET 4.6.2
- [x] .NET Core 1.0.0
- [ ] Something else
### My Environment
- [ ] Windows 7 or below (not truly supported due to EOL)
- [ ] Windows 8
- [ ] Windows 8.1
- [ ] Windows 10
- [ ] Windows 10 IoT Core
- [ ] Windows Server 2012
- [x] Windows Server 2012 R2 (Appveyor)
- [ ] Windows Server 2016
### I have already...
- [x] repeated the problem using the latest stable release of OpenCover.
- [x] reviewed the [usage guide](https://github.com/opencover/opencover/wiki/usage) and [usage document](https://github.com/opencover/opencover/blob/master/main/OpenCover.Documentation/Usage.pdf).
- [x] have looked at the opencover output xml file in an attempt to resolve the issue.
- [x] reviewed the [current issues](https://github.com/opencover/opencover/issues) to check that the issue isn't already known.
### My issue is related to (check only those which apply):
- [x] no coverage being recorded
- [ ] 32 or 64 bit support
- [x] feature request
### Expected Behavior

OpenCover should report some coverage when profiling dotnet core assemblies with portable pdb's.
### Actual Behavior

The assemblies are skipped due to "MissingPdb"
### Steps to reproduce the problem:

Compile a dotnet core assembly with `"debugType": "portable"` in buildOptions in the project.json and profile some tests using OpenCover.

---

Dotnet Core introduced a new format for pdb files (https://github.com/dotnet/core/blob/master/Documentation/diagnostics/portable_pdb.md). OpenCover shoud support this new format when profiling. As stated here https://github.com/OpenCover/opencover/issues/601 Mono.Cecil should provide the needed support. It seems the code is already there: https://github.com/jbevain/cecil/commit/7cde491752861a331e091dca601748e9f0741e95.

**Comment by sawilde** [comment]
>  It seems the code is already there: jbevain/cecil@7cde491.

That code has been there since 31/7/*2015* and OpenCover has been using the package 0.9.6.1 from 30/5/*2015*, however it seems that just yesterday 19/6/*2016* a new package 0.9.6.3 was released; there was no 0.9.6.2.

So it looks like it was released just in time for us to address this feature.

**Comment by jp7677** [comment]
Nice to read that! Looking forward to coverage reports over Dotnet Core C# and F# ;)

**Comment by sawilde** [comment]
Unfortunate 0.9.6.3 does not work and the library is throwing an exception. Will try and investigate but it may not be trivial

**Comment by sawilde** [comment]
Okay it seems the fix is in master for the next release (0.10.*) but is not available for the 0.9.6.* packages

**Comment by jp7677** [comment]
Thanks a lot for sharing you information. I would be happy to make some tests with C# and F# on dotnet core if you manage to release a pre-release on nuget based on the cecil pre-release.

**Comment by sawilde** [comment]
I have a build available here - https://ci.appveyor.com/project/sawilde/opencover/build/4.6.589

The new mono.cecil is due in a week or so, so I'll probably wait for that before pushing out a new OpenCover package.

**Comment by jp7677** [comment]
Hi there. good work! Using your appveyor build I could run my unit tests with portable pdb files and the coverage for C# was reported. Cool!

Unfortunately Opencover still complains about MissingPdb in an F# project with portable pdb's. Is there a way to get some more info from OpenCover about the actual reason? Or could I provide you a sample so that you can step through with a debugger? I could attach some binaries to this issue or alternatively you could take a look at https://github.com/jp7677/hellocoreclr, though the latest master doesn't yet has the "debugType": "portable" switch in the projects.json's.

**Comment by jp7677** [comment]
FYI, updating Microsoft.FSharp.Core.netcore to 1.0.0-alpha-160727 makes no difference. Though of course still possible that the f# coverage issue is related to an issue in f# for .net core and not something within OpenCover.

**Comment by sawilde** [comment]
@jp7677 - A sample would be better as then I am testing the same thing as you are; always hard to replicate these things sometimes.

**Comment by jp7677** [comment]
@sawilde What would work for you? You could 
- clone https://github.com/jp7677/hellocoreclr
- add `"debugType": "portable"` to all three project.json's
- run `dotnet restore`
- run `dotnet test test\app.tests` to validate that everything is fine.
- run 
`OpenCover.Console.exe  -target:dotnet.exe -targetargs:"test test\app.tests" -register:user -filter:"+[HelloCoreClrApp*]* -[HelloCoreClrApp*.Tests*]*" -hideskipped:Filter -mergeoutput -output:reports\coverage-dotnet.xml`
- inspect the `reports\coverage-dotnet.xml` report.

**Comment by jp7677** [comment]
Hi @sawilde I have tested the portable pdb's i.c.w. C# and F# in Dotnet Core 1.1 again with your pre-release again. The C# code was already perfectly covered. It seems that coverage of F# sources is also correctly reported now. Dunno if its DotNet 1.1 or the latest versions of the F# packages for Dotnet Core, but its working now. Very cool ;)
Thus your pre-release with the Cecil beta works perfectly fine now for me for both C# and F#.

**Comment by sawilde** [comment]
Awesome - thanks for the update

**Comment by onovotny** [comment]
Any idea when this will hit nuget.org?

**Comment by sawilde** [comment]
nope - as I have said I am waiting on Mono.Cecil to come out of beta

**Comment by onovotny** [comment]
Okay, any chance you can setup a MyGet feed so we can just reference that until then?

**Comment by axelheer** [comment]
Yeah, or just a beta release on NuGet would be awesome. 🙏

**Comment by onovotny** [comment]
With ppdb's one thing to note is that they can be embedded in the dll. They may not be a standalone dll. 

https://github.com/dotnet/corefx/commit/42422af98831ad9ece1b952f5be9bf25b7bd72dd

**Comment by DaveEmmerson** [comment]
Until this is available does anyone have a workaround for F#? I can set DebugType full in a .csproj and it generates the windows .pdbs, but if I do the same for an .fsproj, no .pdb is produced at all :(

**Comment by jp7677** [comment]
@DaveEmmerson Debug type needs to be "portable" for F#. Furthermore you will need the OpenCover version with the experimental portable-pdb support from this conversation (https://ci.appveyor.com/project/sawilde/opencover/build/4.6.589).

See https://github.com/jp7677/hellocoreclr/blob/master/src/HelloCoreClrApp.Rules/project.json and https://github.com/jp7677/hellocoreclr/blob/master/appveyor.yml, https://github.com/jp7677/hellocoreclr/blob/master/NuGet.Config for a sample.

**Comment by DaveEmmerson** [comment]
@jp7677 thanks. The documentation on the F# side isn't quite right, so they're going to look at making it clear that they only do portable. https://github.com/Microsoft/visualfsharp/issues/2565

I've used your method and it worked fine, cheers 👍

**Comment by sawilde** [comment]
Closing this issue as it is covered by #595 
- raised additional cards to track the other issues raised here

**Comment by manjushavnair** [comment]
Opencover not creating any profling information 
1) Visual Studio 2015 
2) .net 4.5 
3) Latest oepnover 
4) The msbuils target getting executed with our error , but getting a portable  file issue 
5) coverage report file ,coverage history file ..etc getting created . But no content in the files execpet a few line 

here is my target which gets executed with a issue stating that the portable is missing 

 <Target Name="CoverageOC" DependsOnTargets="TestResultsReport">
    <Message Text="CoverageOC called $(MSBuildStartupDirectory)\$(NUnitResultsFolder)  " Importance="high" />
    <Exec Command="..\MVCApp\packages\OpenCover.4.6.519\tools\OpenCover.Console.exe -register:Path64 -enableperformancecounters -searchdirs:&quot;..\MVCApp\packages\NUnit.ConsoleRunner.3.7.0\tools&quot; -targetdir:&quot;..\MVCApp\packages\NUnit.ConsoleRunner.3.7.0\tools&quot; -target:&quot;..\MVCApp\packages\NUnit.ConsoleRunner.3.7.0\tools\nunit3-console.exe&quot; -targetargs:&quot;@(TestAssembly) /version /shadowcopy /trace=Debug /config=Debug /result=..\$(NUnitResultsFolder)\TestResult.xml&quot;    -filter:&quot;+[*.*]*&quot; -output:&quot;..\CoverageResults\coverage-results.xml&quot;" />
    <Message Text="CoverageOC Completed " />
  </Target>

**Comment by manjushavnair** [comment]
<Target Name="CoverageOC" DependsOnTargets="TestResultsReport">
    <Message Text="CoverageOC called $(MSBuildStartupDirectory)\$(NUnitResultsFolder)  " Importance="high" />
    <Exec Command="..\MVCApp\packages\OpenCover.4.6.519\tools\OpenCover.Console.exe -register:Path64 -enableperformancecounters -searchdirs:&quot;..\MVCApp\packages\NUnit.ConsoleRunner.3.7.0\tools&quot; -targetdir:&quot;..\MVCApp\packages\NUnit.ConsoleRunner.3.7.0\tools&quot; -target:&quot;..\MVCApp\packages\NUnit.ConsoleRunner.3.7.0\tools\nunit3-console.exe&quot; -targetargs:&quot;@(TestAssembly) /version /shadowcopy /trace=Debug /config=Debug /result=..\$(NUnitResultsFolder)\TestResult.xml&quot;    -filter:&quot;+[*.*]*&quot; -output:&quot;..\CoverageResults\coverage-results.xml&quot;" />
    <Message Text="CoverageOC Completed " />
  </Target>

**Comment by manjushavnair** [comment]
> 

<Target

 Name="CoverageOC" DependsOnTargets="TestResultsReport">
    <Message Text="CoverageOC called $(MSBuildStartupDirectory)\$(NUnitResultsFolder)  " Importance="high" />
    <Exec Command="..\MVCApp\packages\OpenCover.4.6.519\tools\OpenCover.Console.exe -register:Path64 -enableperformancecounters -searchdirs:&quot;..\MVCApp\packages\NUnit.ConsoleRunner.3.7.0\tools&quot; -targetdir:&quot;..\MVCApp\packages\NUnit.ConsoleRunner.3.7.0\tools&quot; -target:&quot;..\MVCApp\packages\NUnit.ConsoleRunner.3.7.0\tools\nunit3-console.exe&quot; -targetargs:&quot;@(TestAssembly) /version /shadowcopy /trace=Debug /config=Debug /result=..\$(NUnitResultsFolder)\TestResult.xml&quot;    -filter:&quot;+[*.*]*&quot; -output:&quot;..\CoverageResults\coverage-results.xml&quot;" />
    <Message Text="CoverageOC Completed " />
  </Target> 

>

**Comment by manjushavnair** [comment]
https://github.com/manjushavnair/MVCDOTNET/blob/master/MVCAppTests/MVCAppTests.csproj 

I have my Target Name="CoverageOC" which works but shows an error that portable is missing

**Comment by sawilde** [comment]
@manjushavnair please raise this as a new issue rather than hijack a new one - thanks
